pkg/k8s/network_policy.go

```diff
@@ -135,8 +142,13 @@ func ParseNetworkPolicy(np *networkingv1.NetworkPolicy) (api.Rules, error) {
 		if eRule.To != nil && len(eRule.To) > 0 {
 			for _, rule := range eRule.To {
 				if rule.NamespaceSelector != nil || rule.PodSelector != nil {
-					// TODO: GH-2095
-					log.Warning("Cilium does not support PodSelector or NamespaceSelector for K8s Egress rules")
+					endpointSelector := parseNetworkPolicyPeer(namespace, &rule)
+
+					if endpointSelector != nil {
+						egress.ToEndpoints = append(egress.ToEndpoints, *endpointSelector)
+					} else {
```

Comment:
Shouldn't this `else` be for `if rule.NamespaceSelector != nil || rule.PodSelector != nil {`?

Reply:
It looks like `parseNetworkPolicyPeer` can only return nil if both `rule.NamespaceSelector` and `rule.PodSelector` are nil. Can we return an error and reject the policy, it will clearly not do what we want so we should reject it an log the error.

Reply:
@aanm  @tgraf Should this be done for ingress rules as well?

Reply:
@aanm @tgraf ping

Reply:
Yes, same for ingress. We can do this in a separate PR though. I'm fine to keep ingress and egress consistent for now.

Reply:
Filed https://github.com/cilium/cilium/issues/3145 for tracking separately.